src/fielddef.ts

```diff
@@ -250,6 +252,14 @@ export function defaultType(fieldDef: FieldDef, channel: Channel): Type {
 export function normalize(fieldDef: ChannelDef, channel: Channel) {
   // If a fieldDef contains a field, we need type.
   if (isFieldDef(fieldDef)) { // TODO: or datum
+    // TODO: swojit - you can normalize bin here
+    if (fieldDef.bin) {
+      fieldDef.bin = isBoolean(fieldDef.bin) ? {maxbins: 10} : fieldDef.bin;
```

Comment:
This is still causing side effect to the existing bin object.  As I show below for type you should do something like:
```
fieldDef = {
   ...fieldDef,
   bin: {
    maxbins: 10,
    ...(isBoolean(fieldDef.bin) ? {} : fieldDef.bin)
   }
}
```